batch/batch/driver/main.py

```diff
@@ -627,8 +630,11 @@ async def check(tx):
 FROM batches
 LEFT JOIN (
   SELECT batch_id, SUM(`usage` * rate) AS cost
-  FROM aggregated_batch_resources
-  INNER JOIN resources ON aggregated_batch_resources.resource = resources.resource
+  FROM (
+    SELECT batch_id, resource, SUM(`usage`) AS `usage`
+    FROM aggregated_batch_resources
+    GROUP BY batch_id, resource) AS abr
+  INNER JOIN resources ON abr.resource = resources.resource
```

Comment:
I don't think this query needs to change. Each entry in aggregated_batch_resources which shares a resource will get the same `rate` and multiplication distributes through addition.

Reply:
I ran it too and the version without a subquery is substantially faster.

```
mysql> explain SELECT batch_id, SUM(`usage` * rate) AS cost
    -> FROM (
    ->   SELECT batch_id, resource, SUM(`usage`) AS `usage`
    ->   FROM aggregated_batch_resources
    ->   GROUP BY batch_id, resource) AS abr
    -> INNER JOIN resources ON abr.resource = resources.resource
    -> GROUP BY batch_id
    -> ;
+----+-------------+----------------------------+------------+-------+------------------+-------------+---------+--------------------------+--------+----------+---------------------------------+
| id | select_type | table                      | partitions | type  | possible_keys    | key         | key_len | ref                      | rows   | filtered | Extra                           |
+----+-------------+----------------------------+------------+-------+------------------+-------------+---------+--------------------------+--------+----------+---------------------------------+
|  1 | PRIMARY     | resources                  | NULL       | ALL   | PRIMARY          | NULL        | NULL    | NULL                     |      7 |   100.00 | Using temporary; Using filesort |
|  1 | PRIMARY     | <derived2>                 | NULL       | ref   | <auto_key0>      | <auto_key0> | 302     | batch.resources.resource |   2431 |   100.00 | NULL                            |
|  2 | DERIVED     | aggregated_batch_resources | NULL       | index | PRIMARY,resource | PRIMARY     | 310     | NULL                     | 243141 |   100.00 | NULL                            |
+----+-------------+----------------------------+------------+-------+------------------+-------------+---------+--------------------------+--------+----------+---------------------------------+
3 rows in set, 1 warning (0.01 sec)

mysql> explain SELECT batch_id, SUM(`usage` * rate) AS cost
    -> FROM aggregated_batch_resources
    -> INNER JOIN resources ON aggregated_batch_resources.resource = resources.resource
    -> GROUP BY batch_id
    -> ;
+----+-------------+----------------------------+------------+------+------------------+----------+---------+--------------------------+-------+----------+---------------------------------+
| id | select_type | table                      | partitions | type | possible_keys    | key      | key_len | ref                      | rows  | filtered | Extra                           |
+----+-------------+----------------------------+------------+------+------------------+----------+---------+--------------------------+-------+----------+---------------------------------+
|  1 | SIMPLE      | resources                  | NULL       | ALL  | PRIMARY          | NULL     | NULL    | NULL                     |     7 |   100.00 | Using temporary; Using filesort |
|  1 | SIMPLE      | aggregated_batch_resources | NULL       | ref  | PRIMARY,resource | resource | 302     | batch.resources.resource | 40210 |   100.00 | NULL                            |
+----+-------------+----------------------------+------------+------+------------------+----------+---------+--------------------------+-------+----------+---------------------------------+
2 rows in set, 1 warning (0.01 sec)
```

Reply:
(I realize this is a check loop, but we should keep everything fast if its easy!)

Reply:
Two things.

1. I can't do that because we need the `msec_mcpu` field in the batches table.
2. I think we should delete this check because we can't use it anymore. This was only able to be used before we switched to local ssds

Reply:
Oh, I see what you were saying. I'm still going to delete it because we can't use the check anymore.